language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/WaitFieldsListNodeContext.java

```diff
@@ -44,14 +54,50 @@ public WaitFieldsListNodeContext() {
     @Override
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, WaitFieldsListNode node)
             throws LSCompletionException {
-        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
-        List<Symbol> filteredSymbols = visibleSymbols.stream()
-                .filter(symbol -> symbol.kind() == SymbolKind.VARIABLE
-                        && ((VariableSymbol) symbol).typeDescriptor().typeKind() == TypeDescKind.FUTURE)
-                .collect(Collectors.toList());
-        List<LSCompletionItem> completionItems = this.getCompletionItemList(filteredSymbols, context);
+
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
+
+        if (QNameReferenceUtil.onQualifiedNameIdentifier(context, nodeAtCursor)) {
+            completionItems.addAll(this.getExpressionsCompletionsForQNameRef(context, nodeAtCursor));
+        } else {
+            completionItems.addAll(this.expressionCompletions(context));
+        }
         this.sort(context, node, completionItems);
 
         return completionItems;
     }
+    
+    protected List<LSCompletionItem> expressionCompletions(BallerinaCompletionContext context) {
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition());
+        List<Symbol> filteredList = visibleSymbols.stream() 
+                .filter(CommonUtil.getVariableFilterPredicate().or(symbol -> symbol.kind() == WORKER))
+                .collect(Collectors.toList());
+        return this.getCompletionItemList(filteredList, context);
+    }
+
+    protected List<LSCompletionItem> getExpressionsCompletionsForQNameRef(BallerinaCompletionContext context,
+                                                                          NonTerminalNode nodeAtCursor) {
+        QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nodeAtCursor;
+        Predicate<Symbol> filter = symbol -> symbol instanceof VariableSymbol
```

Comment:
Is there a particular reason to skip types. Would it be incorrect to use the `QnameReferenceUtil.getExpressionCompletions()` API

Reply:
There was no particular reason. Will use it instead 👍